Drop-In/src/main/java/com/braintreepayments/api/SelectPaymentMethodParentFragment.java

```diff
@@ -0,0 +1,123 @@
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentResultListener;
+import androidx.viewpager2.widget.ViewPager2;
+
+import com.braintreepayments.api.dropin.R;
+
+import java.util.ArrayList;
+
+public class SelectPaymentMethodParentFragment extends Fragment {
+
+    private ViewPager2 viewPager;
+    private SelectPaymentMethodChildFragmentAdapter viewPagerAdapter;
+    private SelectPaymentMethodChildFragmentList childFragmentList;
+
+    private DropInRequest dropInRequest;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.bt_fragment_select_payment_method_parent, container, false);
+
+        viewPager = view.findViewById(R.id.view_pager);
+        viewPager.setUserInputEnabled(false);
+
+        childFragmentList = new SelectPaymentMethodChildFragmentList(
+                SelectPaymentMethodChildFragment.SUPPORTED_PAYMENT_METHODS);
+
+        Bundle args = getArguments();
+        if (args != null) {
+            dropInRequest = args.getParcelable("EXTRA_DROP_IN_REQUEST");
+        }
+
+        FragmentManager childFragmentManager = getChildFragmentManager();
+        childFragmentManager.setFragmentResultListener(DropInEvent.REQUEST_KEY, this, new FragmentResultListener() {
+            @Override
+            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle result) {
+                onDropInEvent(DropInEvent.fromBundle(result));
+            }
+        });
+
+        viewPagerAdapter = new SelectPaymentMethodChildFragmentAdapter(childFragmentManager, getLifecycle(), childFragmentList, dropInRequest);
+        viewPager.setAdapter(viewPagerAdapter);
+
+        // disable animation for now
+        viewPager.setPageTransformer(new ViewPager2.PageTransformer() {
+            @Override
+            public void transformPage(@NonNull View page, float position) {
+                if (position < -1 || position > 1) {
+                    // page is either offscreen to the left or offscreen to the right
+                    page.setAlpha(0.0f);
+                } else {
+                    // page is visible
+                    page.setAlpha(1.0f);
+                }
+            }
+        });
+
+        return view;
+    }
+
+    @VisibleForTesting
+    void onDropInEvent(DropInEvent event) {
+        switch (event.getType()) {
+            case SHOW_VAULT_MANAGER:
+                onShowVaultManager(event);
+                break;
+            case DISMISS_VAULT_MANAGER:
+                onDismissVaultManager(event);
+                break;
+        }
+
+        // propagate event up to the parent activity
+        sendDropInEvent(event);
```

Comment:
I think we should have fragment UI test suite or unit tests to cover the event handling / propagating to parent activity if possible

Reply:
Done!